### ISSUE TITLE
Support passing a vector of objects in Move calls

### ISSUE THREAD

**Issue by lxfind** [MystenLabs/sui#492]
Currently a Move order only supports a list of individual objets as each object parameter. We should also support having a vector of objects as individual parameter.

**Comment by lxfind** [comment]
> Will take a look at this I reckon I just have the leverage the new `Collection` type right?

Just saw the comment. For this we should be using the Move native vector type.

**Comment by oxade** [comment]
Depends on https://github.com/MystenLabs/sui/issues/1521

**Comment by oxade** [comment]
Spoke to @tnowacki  and we have a general plan for this

**Comment by oxade** [comment]
Handing off to @tnowacki who has more context on adapter side https://github.com/MystenLabs/sui/issues/1521

**Comment by oxade** [comment]
Major areas that need to be updated:
1. RPC: we need to modify the RPC layer to take a vec of object IDs
2. CLI: vectors of object IDs will probably need to be fleshed out for better UX
3. Adapter: we make assumptions of the number of objects going into a TX being the number of objects in the fn signature, but we cannot guarantee this with objects.

Other things of note:
1. Error reporting: when an inner object in a vec has an error for example due to locking or type mismatch, we need to propagate that clearly
2. The checker needs to be improved to check all items in the vector quickly since vectors can be large

**Comment by bholc646** [comment]
@lxfind @tnowacki I think we just said in our planning meeting that this is a duplicate of #1521 but this discussion from 3+ months ago makes it sound like it is different and in fact dependent on #1521. Can someone clarify?

**Comment by lxfind** [comment]
It's pretty much the same thing, can definitely merge
